File: R/Hanoi/hanoi.r
hanoi <- function(n, from, to, via) {
  if (n > 1) {
    hanoi(n - 1, from, via, to)
    cat(paste(from, "->", to, "\n"))
    hanoi(n - 1, via, to, from)
  } else {
    cat(paste(from, "->", to, "\n"))
  }
}

n <- readLines("stdin")
hanoi(as.integer(n), "a", "b", "c")

File: Pascal/Hanoi/hanoi.pas
program Hanoi;

procedure hanoi(n: integer; from: string; _to: string; via: string);
begin
   if n > 1 then
   begin
      hanoi(n - 1, from, via, _to);
      writeln(from + ' -> ' + _to);
      hanoi(n - 1, via, _to, from)
   end
   else
      writeln(from + ' -> ' + _to)
end;

var n: integer;
begin
   readln(n);
   hanoi(n, 'a', 'b', 'c')
end.
